src/measure_theory/measure_space.lean

```diff
@@ -1336,6 +1382,45 @@ lemma finite_at_filter_of_finite (μ : measure α) [finite_measure μ] (f : filt
 lemma measure.finite_at_bot (μ : measure α) : μ.finite_at_filter ⊥ :=
 ⟨∅, mem_bot_sets, by simp only [measure_empty, with_top.zero_lt_top]⟩
 
+/-- A set `s` is called σ-finite w.r.t. measure `μ` if there is a countable collection of sets
+  `{ A i | i ∈ ℕ }` such that `μ (A i) < ⊤` and `⋃ i, A i = s`.
+  A measure `μ` is called σ-finite if `univ` is σ-finite w.r.t. `μ`.
+  Note that this class is not a Proposition.
```

Comment:
Why don't you want it to be a `Prop`? I'm afraid that this will create some problems with non-defeq instances later. Am I right that `Prop`-version is equivalent to `∃ t : ℕ → set α, (⋃ i, t i) = s ∧∀ i, μ (t i) < ⊤`? BTW, what do you think about using `sigma_finite (μ.restrict s)` for `s ≠ univ`?

Reply:
Those are great points! I will delete `sigma_finite` for now, and will make a new PR with the `sigma_finite` stuff.